nucypher/blockchain/eth/networks.py

```diff
@@ -1,57 +1,83 @@
-class NetworksInventory:  # TODO: See #1564
-
-    MAINNET = "mainnet"
-    LYNX = "lynx"
-    IBEX = "ibex"  # this is required for configuration file migrations (backwards compatibility)
-    ETH = "ethereum"
-    TAPIR = "tapir"
-    ORYX = "oryx"
-
-    # TODO: Use naming scheme to preserve multiple compatibility with multiple deployments to a single network?
-    POLYGON = 'polygon'
-    MUMBAI = 'mumbai'
-
-    UNKNOWN = 'unknown'  # TODO: Is there a better way to signal an unknown network?
-    DEFAULT = MAINNET
-
-    __to_chain_id_eth = {
-        MAINNET: 1,  # Ethereum Mainnet
-        IBEX: 5,  # this is required for configuration file migrations (backwards compatibility)
-        LYNX: 5,  # Goerli
-        TAPIR: 5,
-        ORYX: 5
-    }
-    __to_chain_id_polygon = {
-        # TODO: Use naming scheme?
-        POLYGON: 137,    # Polygon Mainnet
-        MUMBAI: 80001,   # Polygon Testnet (Mumbai)
-    }
-
-    ETH_NETWORKS = tuple(__to_chain_id_eth.keys())
-    POLY_NETWORKS = tuple(__to_chain_id_polygon.keys())
-
-    NETWORKS = ETH_NETWORKS + POLY_NETWORKS
-
-    class UnrecognizedNetwork(RuntimeError):
-        pass
+from enum import Enum
+from typing import List, NamedTuple
 
-    @classmethod
-    def get_ethereum_chain_id(cls, network):
-        try:
-            return cls.__to_chain_id_eth[network]
-        except KeyError:
-            raise cls.UnrecognizedNetwork(network)
+from nucypher.config.constants import TEMPORARY_DOMAIN
+
+
+class ChainInfo(NamedTuple):
+    chain_id: int
+    chain_name: str
+
+    # Override eventual Enum name. TODO: better way?
+    @property
+    def name(self) -> str:
+        return self.chain_name
+
+
+class EthNetwork(ChainInfo, Enum):
+    MAINNET = ChainInfo(1, "mainnet")
+    GOERLI = ChainInfo(5, "goerli")
+    SEPOLIA = ChainInfo(11155111, "sepolia")
+    # testing
+    TESTERCHAIN = ChainInfo(131277322940537, TEMPORARY_DOMAIN)
+
+
+class PolyNetwork(ChainInfo, Enum):
+    POLYGON = ChainInfo(137, "polygon")
+    MUMBAI = ChainInfo(80001, "mumbai")
+    # testing
+    TESTERCHAIN = ChainInfo(131277322940537, TEMPORARY_DOMAIN)
+
+
+class TACoNetwork(NamedTuple):
+    name: str
+    eth_network: EthNetwork
+    poly_network: PolyNetwork
+
+    def is_testnet(self) -> bool:
+        return self.eth_network != EthNetwork.MAINNET
+
+
+class UnrecognizedNetwork(RuntimeError):
+    """Raised when a provided network name is not recognized."""
+    pass
```

Comment:
`pass` can be removed here - the docstring is enough to indicate a block.

```suggestion
```

Reply:
✔️